packages/transformers/js/core/src/lib.rs

```diff
@@ -388,6 +390,21 @@ pub fn transform(config: Config) -> Result<TransformResult, std::io::Error> {
               );
               result.has_node_replacements = has_node_replacements;
 
+              // Flush (JsWord, SyntaxContexts) into unique names and reresolve to
+              // set global_mark for all nodes, even generated ones.
+              // - This changes the syntax context ids and therefore invalidates decls
+              // - This will also remove any other other marks (like ignore_mark)
+              let (decls, module) = if config.scope_hoist {
+                let module = module.fold_with(&mut chain!(
+                  hygiene(),
+                  resolver(unresolved_mark, global_mark, false)
+                ));
+                (collect_decls(&module), module)
```

Comment:
So if the node replacements pass doesn't run we do this twice in a row? Also, are there more cases where these extra passes can be skipped (e.g. if no preset-env)?

Reply:
I've made our replacement passes use the "incorrect" global syntax context directly (which is what will be the case after the resolver reruns anyway). So now it's only needed for preset_env.

`collect_decls` has to always be rerun here because of line 3:
```
              // Flush (JsWord, SyntaxContexts) into unique names and reresolve to
              // set global_mark for all nodes, even generated ones.
              // - This changes the syntax context ids and therefore invalidates decls
```

Reply:
hmm I guess the real question is why the first one exists before the node replacements pass, or why the node pass runs after preset-env rather than before?